src/twisted/mail/imap4.py

```diff
@@ -209,33 +275,76 @@ def clean(self):
         Clean ranges list, combining adjacent ranges
         """
 
-        self.ranges.sort()
+        ranges = sorted((self._swap(low, self._infinity, ifIs=None),
+                         self._swap(high, self._infinity, ifIs=None))
+                         for low, high in self.ranges)
 
-        oldl, oldh = None, None
-        for i,(l, h) in enumerate(self.ranges):
-            if l is None:
+        mergedRanges = [(float('-inf'), float('-inf'))]
```

Comment:
I left these as literals because they're used only in this function as the base case for the loop that follows, whereas `MessageSet._infinity`'s unique identity allows `_swap` to reliably identify it.